### DISCUSSION TITLE
`try`-`finally` (tweet)

### DISCUSSION THREAD

**Discussion by ggreif** [dfinity/motoko#4817] [Show and tell]
The recent addition of `finally`-clauses to `try`-blocks not only gives you a powerful tool to clean up (e.g. locks) in a disciplined and structured way. It will interpose all control-flow paths outward of the `try`-`catch` and _additionally_ also clean up if a canister traps after receiving a reply from extern. Previously this scenario could potentially brick your canister. Still, be very careful when reviewing code that does external messaging and may trap. Continue learning at https://internetcomputer.org/docs/current/motoko/main/writing-motoko/errors/#using-tryfinally.

**Comment by jessiemongeon1** [comment]
Posted on twitter: https://x.com/DFINITYDev/status/1900481277649629622
